shared/middleware/net_sim_api.rb

```diff
@@ -78,6 +119,21 @@ def has_json_utf8_headers(request)
     dont_cache
     table = get_table(shard_id, table_name)
     int_id = id.to_i
+
+    if table_name == TABLE_NAMES[:node]
+      # Cascade deletions
+      wire_ids = delete_wires_for_node(shard_id, int_id)
+      message_ids = delete_messages_for_node(shard_id, int_id)
+
+      unless wire_ids.empty?
+        get_pub_sub_api.publish(shard_id, TABLE_NAMES[:wire], {:action => 'delete_many', :ids => wire_ids})
```

Comment:
I have a related comment on your PR :grin:.  Here we've at least gone from one publish-per-delete to one publish-per-affected-table.  We explicitly called out in the PR description that we're holding further optimization here for a future PR.

You're correct that we eventually want a maximum of one publish event per HTTP request, where the event data can enumerate which tables and IDs were affected.  That will require some new intelligence on the client though, to parse the event data and make the correct read request(s).